easybuild/easyblocks/generic/rubygem.py

```diff
@@ -0,0 +1,89 @@
+##
+# Copyright 2009-2013 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://vscentrum.be/nl/en),
+# the Hercules foundation (http://www.herculesstichting.be/in_English)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for Ruby Gems, implemented as an easyblock
+
+@author: Robert Schmidt (OHRI)
+"""
+
+from easybuild.framework.extensioneasyblock import ExtensionEasyBlock
+
+from easybuild.tools.run import run_cmd
+
+class RubyGem(ExtensionEasyBlock):
+    """ Builds and installs Ruby Gems """
+
+    def __init__(self, *args, **kwargs):
+        super(RubyGem, self).__init__(*args, **kwargs)
+
+    def install_ruby_gem(self):
+        """install ruby gems using gem package manager"""
+        run_cmd('gem install %s' % (self.ext_src))
+
+    def run(self):
+        
+        if not self.src:
+            self.log.error("No source found for Ruby Gem %s, required for installation. (src: %s)" %
+                           (self.name, self.src))
+        super(RubyGem, self).run()
+        if self.src:
```

Comment:
I was basing this from R and Perl and there are some slight differences in the interface. It seems like self.src is true, but self.ext_src is empty and needs to be replace. I'll pick at it some more.